lib/postgres.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Library made to simplify accessing / connecting to postgres databases,
+ * and to cleanly handle when the pg module isn't installed.
+ * @author mia-pi-git
+ */
+
+import type {Pool as PostgresPool} from 'pg';
+import type {SQLStatement} from 'sql-template-strings';
+
+const DEFAULT_CONFIG = 'Config' in global ? Config.usepostgres : null;
+
+export class PostgresDatabase {
+	private pool: PostgresPool;
+	connected = false;
+	constructor(config = DEFAULT_CONFIG) {
+		this.pool = config ? new (require('pg').Pool)(config) : null!;
```

Comment:
Weren't we going to try using Odyssey to pool outside PS?

Reply:
I think so? but i set it as a pool for now since i was confused whether you were sure on that or not.